qcodes/instrument_drivers/tektronix/keithley_7510.py

```diff
@@ -1,10 +1,367 @@
+        self._size = size
+        self.style = style
+        self.data_start = 1  # first index of the data to be returned
+        self.data_end = 1    # last index of the data to be returned
+
+        if self.short_name not in self.default_buffer:
+            # when making a new buffer, the "size" parameter is required.
+            if size is None:
+                raise TypeError(
+                    "buffer() missing 1 required positional argument: 'size'"
+                )
+            self.write(
+                f":TRACe:MAKE '{self.short_name}', {self._size}, {self.style}"
+            )
+        else:
+            # when referring to default buffer, "size" parameter is not needed.
+            if size is not None:
+                self.log.warning(
+                    f"Please use method 'size()' to resize default buffer "
+                    f"{self.short_name} size to {self._size}."
+                )
+
+        self.add_parameter(
+            "size",
+            get_cmd=f":TRACe:POINts? '{self.short_name}'",
+            set_cmd=f":TRACe:POINts {{}}, '{self.short_name}'",
+            get_parser=int,
+            docstring="The number of readings a buffer can store."
+        )
+
+        self.add_parameter(
+            "number_of_readings",
+            get_cmd=f":TRACe:ACTual? '{self.short_name}'",
+            get_parser=int,
+            docstring="Get the number of readings in the reading buffer."
+        )
+
+        self.add_parameter(
+            "last_index",
+            get_cmd=f":TRACe:ACTual:END? '{self.short_name}'",
+            get_parser=int,
+            docstring="Get the last index of readings in the reading buffer."
+        )
+
+        self.add_parameter(
+            "first_index",
+            get_cmd=f":TRACe:ACTual:STARt? '{self.short_name}'",
+            get_parser=int,
+            docstring="Get the starting index of readings in the reading "
+                      "buffer."
+        )
+
+        self.add_parameter(
+            "elements",
+            get_cmd=None,
+            get_parser=self._from_scpi_to_name,
+            set_cmd=None,
+            set_parser=self._from_name_to_scpi,
+            vals=Lists(Enum(*list(self.buffer_elements.keys()))),
+            docstring="List of buffer elements to read."
+        )
+
+        self.add_parameter(
+            "setpoints_start",
+            label="start value for the setpoints",
+            source=None,
+            parameter_class=DelegateParameter
+        )
+
+        self.add_parameter(
+            "setpoints_stop",
+            label="stop value for the setpoints",
+            source=None,
+            parameter_class=DelegateParameter
+        )
+
+        self.add_parameter(
+            "n_pts",
+            label="total n for the setpoints",
+            unit="",
+            initial_value=10,
+            get_cmd=None,
+            set_cmd=None,
+            get_parser=int
+        )
```

Comment:
When data_start and and data_end are made parameters this can simply be made a parameter that can be get but not set and calculates the number of setpoints from the other paramters

Reply:
Good point. I have made the changes.